samcli/lib/deploy/deployer.py

```diff
@@ -481,15 +499,20 @@ def wait_for_execute(self, stack_name: str, stack_operation: str, disable_rollba
             waiter.wait(StackName=stack_name, WaiterConfig=waiter_config)
         except botocore.exceptions.WaiterError as ex:
             LOG.debug("Execute stack waiter exception", exc_info=ex)
-            if disable_rollback:
+            if disable_rollback and on_failure is not FailureMode.DELETE:
```

Comment:
Make sure to clarify that the only valid case here is when the failure mode is DO_NOTHING or disable_rollback is set to True